packages/@jsii/go-runtime/jsii-runtime-go/internal/typeregistry/registration.go

```diff
@@ -150,3 +151,20 @@ func (t *TypeRegistry) RegisterStruct(fqn api.FQN, strct reflect.Type) error {
 
 	return nil
 }
+
+// RegisterStructValidator adds a validator function to an already registered struct type. This is separate call largely
+// to maintain backwards compatibility with existing code.
```

Comment:
Why do we need to emit this separate call to maintain backwards compatibility, instead of just bundling this into the struct emitter?

Reply:
In go, structs are "pure data", so there is no option to validate them at creation... In order to work around this, I have the go runtime library register validation functions for each struct type...

I need a call into the runtime library to actually inform it about a validation function... But I cannot add a new argument to `RegisterStruct` because this would break code generated against and older runtime library... so this new feature uses a new runtime call.